dockstore-client/src/main/java/io/github/collaboratory/wdl/WDLClient.java

```diff
@@ -170,92 +174,123 @@ public long launch(String entry, boolean isLocalEntry, String yamlRun, String js
-                        if (inputJson.containsKey(outFile + ".metadata")) {
-                            byte[] metadatas = Base64.getDecoder().decode((String)inputJson.get(outFile + ".metadata"));
-                            new1.setMetadata(new String(metadatas, StandardCharsets.UTF_8));
+                        // grab values from output JSON
+                        Map<String, String> outputJson = gson.fromJson(bracketContents, HashMap.class);
+                        System.out.println("Provisioning your output files to their final destinations");
+                        final List<String> outputFiles = bridge.getOutputFiles(localPrimaryDescriptorFile);
+                        FileProvisioning fileProvisioning = new FileProvisioning(abstractEntryClient.getConfigFile());
+                        List<ImmutablePair<String, FileProvisioning.FileInfo>> outputList = new ArrayList<>();
+                        for (String outFile : outputFiles) {
+                            // find file path from output
+                            final File resultFile = new File(outputJson.get(outFile));
+                            FileProvisioning.FileInfo new1 = new FileProvisioning.FileInfo();
+                            new1.setUrl(wdlOutputTarget + "/" + outFile);
+                            new1.setLocalPath(resultFile.getAbsolutePath());
+                            if (inputJson.containsKey(outFile + ".metadata")) {
+                                byte[] metadatas = Base64.getDecoder().decode((String)inputJson.get(outFile + ".metadata"));
+                                new1.setMetadata(new String(metadatas, StandardCharsets.UTF_8));
+                            }
+                            System.out.println("Uploading: " + outFile + " from " + resultFile + " to : " + new1.getUrl());
+                            outputList.add(ImmutablePair.of(resultFile.getAbsolutePath(), new1));
                         }
-                        System.out.println("Uploading: " + outFile + " from " + resultFile + " to : " + new1.getUrl());
-                        outputList.add(ImmutablePair.of(resultFile.getAbsolutePath(), new1));
+                        fileProvisioning.uploadFiles(outputList);
+                    } else {
+                        System.out.println("Output files left in place");
                     }
-                    fileProvisioning.uploadFiles(outputList);
-                } else {
-                    System.out.println("Output files left in place");
+                } catch (Exception e) {
+                    notificationsClient.sendMessage(NotificationsClient.PROVISION_OUTPUT, false);
+                    throw e;
+                }
+            } else {
+                WorkflowExecutionServiceApi clientWorkflowExecutionServiceApi = abstractEntryClient.getWorkflowExecutionServiceApi(wesUrl);
+                try {
+                    String tags = "WorkflowExecutionService";
```

Comment:
Can use this inline on line 289 unless it is used often, in which case a constant is more appropriate.